### ISSUE TITLE
check_log does not check for "-O oldlog"

### ISSUE THREAD

**Issue by box293** [nagios-plugins/nagios-plugins#251]
```
/usr/local/nagios/libexec/check_log -F /var/log/cron

/usr/local/nagios/libexec/check_log: line 187: : No such file or directory
Log check data initialized...
```
Help indicates that -O is a required argument.
```
Usage: check_log -F logfile -O oldlog -q query
```

It works when -O xxxx is used:
```
/usr/local/nagios/libexec/check_log -F /var/log/cron -O /var/log/cron-20170313 

Log check ok - 0 pattern matches found|match=0;;;0
```

However it doesn't test if -O has a supplied value:
```
/usr/local/nagios/libexec/check_log -F /var/log/cron -O

/usr/local/nagios/libexec/check_log: line 187: : No such file or directory
Log check data initialized...
```

Also, help indicates that "-q query" is a required value but doesn't provide any example of the query syntax.

**Comment by jfrickson** [comment]
I'll look into it

**Comment by jfrickson** [comment]
Fixed in branch `maint` via commit https://github.com/nagios-plugins/nagios-plugins/commit/f45107b8d404e6268c39c7d4cc3b854e14ee18a7

Will be released in version 2.2.2

**Comment by meabandit** [comment]
I do not think it should be a required parameter, see the following:
```
[root@core ~]# logger fubar
[root@core ~]# /usr/local/nagios/libexec/check_log -F /var/log/messages -O /dev/null -q fubar
(1) < May 11 12:49:17 core root: fubar|match=1;;;0
```
IMO it would be more logical to remove that parameter or make it optional and work if not given.
